src/main/app/forms/validation/validators/fractions.ts

```diff
@@ -29,6 +29,10 @@ export class FractionsConstraint implements ValidatorConstraintInterface {
       return true
     }
 
+    if (value < 0) {
```

Comment:
refactor this validator. Too many return trues. Combine in one method or use a switch statement.